tests/Api/PersistentSpec.hs

```diff
@@ -41,18 +41,25 @@ getDbSpec name getXs = do
                 xs <- rp getXs
                 length xs `shouldNotBe` 0
 
-addDbSpec :: (Foldable f, Arbitrary proto) => String -> Persist (f a) -> (proto -> Persist a) -> Spec
-addDbSpec name getXs addX =
+addDbSpecProp :: (Foldable f, Arbitrary proto) =>
+                 String -> Persist (f a) -> (proto -> Persist a) -> (proto -> a -> Bool) -> Spec
+addDbSpecProp name getXs addX propX =
     describe name $ do
         let t = it "adds one" $ \(Nat rp) -> do
                     before' <- liftIO $ length <$> rp getXs
-                    liftIO $ generate arbitrary >>= rp . addX
+                    p <- liftIO $ generate arbitrary
+                    r <- liftIO . rp $ addX p
                     after' <- liftIO $ length <$> rp getXs
                     after' `shouldBe` before' + 1
+                    propX p r `shouldBe` True
```

Comment:
(sorry, I meant: `propX :: a -> Expectation`.  Not sure this is what we want, anyway the type of `shouldBe` is `a -> a -> Expectation` with some constraints.)